### ISSUE TITLE
Copy data-* attributes from amp-img onto img

### ISSUE THREAD

**Issue by chvillanuevap** [ampproject/amphtml#24787]
I currently display a hidden image at the bottom of my website for users to pin by using this code:

```html
<div class="hidden-pin-image">
    <img src="image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" data-pin-media="https://thepostmansknock.com/wp-content/uploads/2019/09/1Remy_Hawaii.jpg">
</div>
```
The `src` attribute is a 1x1 blank image, and the `data-pin-media` attribute is the one that holds the actual pinnable image.

I have the Pinterest extension installed on Google Chrome. When I click on the Pinterest logo next to the URL bar in Chrome, Pinterest pulls the 1Remy_Hawaii.jpg image correctly. However, when AMP is activated, the `img` tag gets converted to an `amp-img`, which causes the Pinterest extension to no longer be able to pull the image correctly. I noticed this behavior when I went to check our Pinterest stats, and noticed that our number of pins had decreased.

However, I see that inside `amp-img`, there is a `<i-amphtml-sizer class="i-amphtml-sizer">` piece that contains an `img` tag. But this `img` tag did not carry over the data-pin-media attribute. If I open the developer console, copy and paste the `data-pin-media` information into the HTML, and try to use the Pinterest extension again, then Pinterest does pull the correct image.

Is there a way to let the AMP plugin know that this attribute needs to be carried over to the `img` tag inside `i-amphtml-sizer`?

The other alternative would be for the Pinterest extension to support `amp-img`'s.

**Comment by erwinmombay** [comment]
@nainar seems like a good candidate for the code jam? let me know what you think!

**Comment by nainar** [comment]
I agree!

**Comment by erwinmombay** [comment]
@nainar do you see any downsides propagating all `data-*` attributes on to amp-img?

**Comment by nainar** [comment]
No, in fact we should have done this before. Like making Instagram and pinterest embeds work well.

**Comment by westonruter** [comment]
Originally reported at https://github.com/ampproject/amp-wp/issues/3369

**Comment by chvillanuevap** [comment]
Will I basically see these commits in a future version of Google Chrome? I noticed that the Pinterest button is still not working as expected. Thanks!

**Comment by kevinkimball** [comment]
@chvillanuevap I just confirmed that this code is live. If you are still seeing unexpected behavior, please open an issue documenting it (but feel free to reference this PR as context)

**Comment by chvillanuevap** [comment]
Sure! I can open a new issue. Just want to confirm the problem is not on my end. This is the HTML output I see:
```
<div class="hidden-pin-image">
    <amp-img src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" class="skip-lazy amp-wp-unknown-size amp-wp-unknown-width amp-wp-unknown-height amp-wp-enforced-sizes i-amphtml-element i-amphtml-layout-intrinsic i-amphtml-layout-size-defined" data-pin-media="https://i0.wp.com/thepostmansknock.com/wp-content/uploads/2016/10/beginners_guide_calligraphy_pinterest.jpg?fit=600%2C2102&amp;ssl=1" object-fit="contain" width="960" height="400" layout="intrinsic" i-amphtml-layout="intrinsic" i-amphtml-auto-lightbox-visited="">
        <i-amphtml-sizer class="i-amphtml-sizer">
            <img alt="" role="presentation" aria-hidden="true" class="i-amphtml-intrinsic-sizer" src="data:image/svg+xml;charset=utf-8,<svg height=&quot;400px&quot; width=&quot;960px&quot; xmlns=&quot;http://www.w3.org/2000/svg&quot; version=&quot;1.1&quot;/>">
        </i-amphtml-sizer>
        <noscript>...</noscript>
    </amp-img>
</div>
```
The `data-pin-media` attribute was not copied to the `<img>` tag. Is it because the image is hidden with the class `hidden-pin-image`? I noticed other images on the same page have an additional `img` tag after `noscript` with the classes `i-amphtml-fill-content i-amphtml-replaced-content`, so maybe it's just this hidden image.

**Comment by kevinkimball** [comment]
@chvillanuevap Are there any errors being logged in the console? Just want to check there is nothing obviously wrong happening

**Comment by chvillanuevap** [comment]
No, there are no errors in the console.

**Comment by chvillanuevap** [comment]
Should I go ahead and open a new issue? Thanks!

**Comment by kevinkimball** [comment]
@chvillanuevap I confirmed that it is working as expected for me and the data attributes are copied from the amp-img component to the img element. Try this code: 

```
<amp-img
  src="image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" 
  data-pin-media="https://thepostmansknock.com/wp-content/uploads/2019/09/1Remy_Hawaii.jpg"
  height="1"
></amp-img>
```

I believe the issue is your use of the image, a hidden img element will never be loaded by AMP by design. Please check out the [documentation](https://amp.dev/documentation/guides-and-tutorials/learn/amp-html-layout/?format=websites) for more info.

**Comment by chvillanuevap** [comment]
That makes sense! I added the `height="1"` attribute to the images, and stopped hiding them. They show now for Pinterest. Thanks!
